### PR TITLE
 #474 implement backup_manifest creation

### PR DESCRIPTION
**Pull request by didiermichel**

### CODE REVIEW

File: barman/storage/file_stats.py
Comment (mikewallace1979) [inline barman/storage/file_stats.py:42]: My immediate thought here was "we should use whatever time zone pg_basebackup uses which will surely be the time zone of the PostgreSQL server" however a few tests with PG14 showed that the backup manifest produced by pg_basebackup would use GMT regardless of the PostgreSQL time zone or the OS time zone.

The PG docs don't specify the time zone but [this mailing list discussion](https://postgrespro.com/list/id/4cea0d44-7524-dc7d-9043-8c701ccaba0a@oss.nttdata.com) suggests Last-Modified is always GMT.

Comment (didiermichel) [reply barman/storage/file_stats.py:42]: I indeed noticed that GMT is used in backup manifest. I tested pg_verifybackup with random srting, it is not checked and just informational.
For consistency, I should probably add GMT to formated date.

File: doc/manual/42-server-commands.en.md
Comment (mikewallace1979) [inline doc/manual/42-server-commands.en.md:60]: I think it's worth emphasizing here that we generate the backup_manifest from the files already stored on the barman server, since that is a key difference between a barman-generated backup_manifest and a backup_manifest generated by `pg_basebackup`.

File: tests/requirements_minimal.txt
Comment (mikewallace1979) [inline tests/requirements_minimal.txt:9]: Due to the use of [tz.UTC](https://github.com/EnterpriseDB/barman/pull/475/files#diff-bf2bad4e300610a4cebe48674911e98fa7d7a819b133035fb14e8eff6130a0a3R29) there is a version dependency with python-dateutil and python 2. I'm not sure which version `tz.UTC` was added, but `python-dateutil==2.6.1` (the OS package version on Ubuntu 18.04) does not have it (whereas 2.8.1 does).

Comment (mikewallace1979) [reply tests/requirements_minimal.txt:9]: Ok so it's 2.7.0 which introduced `tz.UTC`: https://dateutil.readthedocs.io/en/stable/tz.html#dateutil.tz.dateutil.tz.UTC

I think this can be resolved by using `tz.tzutc()` in place of `tz.UTC`.

**Review by mikewallace1979 — COMMENTED** [review]
This is coming together nicely - only minor comments from me

File: tests/requirements_minimal.txt
Comment (didiermichel) [reply tests/requirements_minimal.txt:9]: indeed, tz.UTC exist since 2.7.0.  I actually encounter some issues with barman testing. python-dateutil is installed with apt-get wich provides 2.6.1.  trying pip install  `python-dateutil>=2.7.0` does not change anything whereas  `python-dateutil==2.7.0` does the trick but I find it a bit limiting.
Not sure how to solve this properly yet.

I will have a look at tz.tzutc()

File: barman/cli.py
Comment (didiermichel) [reply barman/cli.py:1464]: Good point, I was not sure cmd_aliases was optional.

File: barman/command_wrappers.py
Comment (didiermichel) [reply barman/command_wrappers.py:1031]: I wish, but for some reason it is not always. I barman-testing docker images for example it is not in the PATH. I am not sure what is the best way to handle it.
Probably something to do at installation time. For docker images  do some fix in docker images to behave like exec is in the path 
`ENV PATH=$PATH:/usr/lib/postgresql/$PG_MAJOR/bin`
Hoping there will be no side effect with other pg binary we usually find in `usr/bin/`

What is your view on this?

File: tests/requirements_minimal.txt
Comment (didiermichel) [reply tests/requirements_minimal.txt:9]: So tz.tzutc() seems to work fine. I will use it.

File: barman/command_wrappers.py
Comment (mikewallace1979) [reply barman/command_wrappers.py:1031]: You're right - `postgresql-client-13` provides the `pg_verifybackup` command in `/usr/lib/postgresql/$PG_MAJOR/bin` (along with various other client applications) but it's the `postgresql-client-common` package which adds the `pg_*` commands which Barman uses into `/usr/bin` and `postgresql-client-common` does not include `pg_verifybackup`.

This feels like something which should be fixed at the packaging level, i.e. the barman OS package would make sure pg_verifybackup is in the path (and anyone installing from source would need to make sure it is in the path themselves).

File: doc/manual/42-server-commands.en.md
Comment (didiermichel) [reply doc/manual/42-server-commands.en.md]: Please check if the updated text is enough.

File: barman/command_wrappers.py
Comment (didiermichel) [reply barman/command_wrappers.py:1031]: I have added a temporary fix adding pg_verifybackup path to PATH.
Not sure how to fix this at packaging level. I will check the repo.

Comment (mikewallace1979) [reply barman/command_wrappers.py]: For the benefit of the PR, we discussed this synchronously and concluded we'll leave it up to the user to make sure the right version of the `pg_verifybackup` binary is on the path.

We'll update the installation docs to cover this (and also `pg_receivewal` since that has a similar issue) but that does not need to block this PR.

**Review by mikewallace1979 — APPROVED** [review]

**Pull request merged**
